src/playlist/playlist.cpp

```diff
@@ -1705,14 +1705,25 @@ Song Playlist::current_item_metadata() const {
   return current_item()->Metadata();
 }
 
-void Playlist::UpdateScrobblePoint() {
+void Playlist::UpdateScrobblePoint(qint64 seek_point) {
   const qint64 length = current_item_metadata().length_nanosec();
 
-  if (length == 0) {
-    scrobble_point_ = 240ll * kNsecPerSec;  // 4 minutes
+  if (seek_point == 0) {
+    if (length == 0) {
+      scrobble_point_ = 240ll * kNsecPerSec;  // 4 minutes
```

Comment:
Can you add some more descriptive constants while you're here about what the 240 & 31 actually mean? Something like UNKNOWN_LENGTH_SCROBBLE_POINT_SECS